src/stepfunctions/steps/states.py

```diff
@@ -541,13 +563,13 @@ def __init__(self, state_id, **kwargs):
         Args:
             state_id (str): State name whose length **must be** less than or equal to 128 unicode characters. State names **must be** unique within the scope of the whole state machine.
             iterator (State or Chain): State or chain to execute for each of the items in `items_path`.
-            items_path (str, optional): Path in the input for items to iterate over. (default: '$')
+            items_path (str or Placeholder, optional): Path in the input for items to iterate over. (default: '$')
             max_concurrency (int, optional): Maximum number of iterations to have running at any given point in time. (default: 0)
             comment (str, optional): Human-readable comment or description. (default: None)
-            input_path (str, optional): Path applied to the state’s raw input to select some or all of it; that selection is used by the state. (default: '$')
+            input_path (str or Placeholder, optional): Path applied to the state’s raw input to select some or all of it; that selection is used by the state. (default: '$')
             parameters (dict, optional): The value of this field becomes the effective input for the state.
             result_path (str, optional): Path specifying the raw input’s combination with or replacement by the state’s result. (default: '$')
```

Comment:
not-blocking: Why is result_path omitted?

Reply:
The result_path was omitted as it only works with the $[], but i agree that it can be included in this PR as well, will add that in next commits

Reply:
What's `$[]`? You mean it doesn't work with the context object?

Let's keep it out for now. Just create a separate issue to set resultPath with Placeholders.

For the record: every question question I ask doesn't need to be implemented. You are free (and expected!) to push back if you find a reason. It's best to keep the scope of PRs small anyways

Reply:
Yes, that is correct - it does not work with a context object
> For the record: every question question I ask doesn't need to be implemented. You are free (and expected!) to push back if you find a reason. It's best to keep the scope of PRs small anyways

ACK, I will keep that in mind moving forward